codes/pop.py

```diff
@@ -0,0 +1,154 @@
+import codes.utils as cutil
+import pandas as pd
+import numpy as np
+
+def get_adm_fields(adm_level, field_name='name'):
+    """Get list of adm-fields from `adm_level` up to the adm1 level"""
+    return [f'adm{i}_' + field_name for i in range(1, adm_level + 1)]
+
+def get_adm_pops(adm_level, country_code):
+    """Get all populations at an adm-level within a country
+
+    Args:
+        adm_level (int): Adm-level of requested populations.
+        country_code (str): Three-letter country code of requested populations
+
+    Returns:
+        pandas.DataFrame: List of populations at `adm_level` within country corresponding to `country_code`.
+            Indexed by all levels from "adm1" (first level) up to "adm{`adm_level`}"" (last level)
+
+    """
+    path_adm = cutil.DATA_INTERIM / 'adm' / f'adm{adm_level}' / f'adm{adm_level}.csv'
+    adm_df = pd.read_csv(path_adm)
+    indices = get_adm_fields(adm_level)
+    return (adm_df.loc[adm_df['adm0_name'] == country_code]
+            .set_index(indices)[['population']]
+            .rename(columns={
+                'population':f'adm{adm_level}_pop'
+            }))
+
+def merge_policies_with_population_on_level(policies, adm_level, country_code):
+    """Assign all populations at an adm-level to DataFrame of policies
+
+    Args:
+        policies (pandas.DataFrame): List of policies as formatted in ```data/raw/{`country_code`}/{`country_code`}_policy_data_sources.csv```
+        adm_level (int): Adm-level of requested populations.
+        country_code (str): Three-letter country code of requested populations
+
+    Returns:
+        pandas.DataFrame: `policies` with a new column, "adm_{`adm_level`}_pop"
+
+    """
+    adm_pop = get_adm_pops(adm_level, country_code)
+    policies = pd.merge(
+        policies, 
+        adm_pop, 
+        how='left', 
+        left_on=get_adm_fields(adm_level), 
+        right_index=True,   
+    )
+    return policies
+
+def merge_policies_with_population(policies, country_code, max_adm_level):
```

Comment:
In this func (and in the merge into cases func as well), can you add an `errors` kwarg which can take the value "ignore", "warn", or "raise". And then after all the adm-level merges, we check if every policy or case observation has been successfully matched to a population. If not, we either warn, raise a ValueError, or do nothing?